test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -218,24 +218,34 @@ public HubConnectionTests(ServerFixture<Startup> serverFixture, ITestOutputHelpe
             }
         }
 
-        public static IEnumerable<object []> HubProtocolsXTransports()
+        public static IEnumerable<object[]> HubProtocolsXTransportsXHubPaths()
         {
             foreach (var protocol in HubProtocols)
             {
                 foreach (var transport in TransportTypes())
                 {
-                    yield return new object[] { protocol, transport };
+                    foreach (var hubPath in HubPaths())
+                    {
+                        yield return new object[] { protocol, transport, hubPath };
+                    }
                 }
             }
         }
 
+        public static IEnumerable<string> HubPaths()
+            {
+                yield return "/default";
+                yield return "/dynamic";
+            }
```

Comment:
This can just be a static readonly array: `public static string[] HubPaths = new [] { "/default", "/dynamic" }`. Generally `yield` should be used when the values are dynamically calculated (I say this knowing full well I've broken this "rule" once or twice myself ;P)